receiver/hostmetricsreceiver/hostmetrics_linux.go

```diff
@@ -0,0 +1,58 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux
+
+package hostmetricsreceiver
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+func validateRootPath(rootPath string) error {
+	if rootPath == "" {
+		return nil
+	}
+
+	if _, err := os.Stat(rootPath); err != nil {
+		return fmt.Errorf("invalid root_path: %w", err)
+	}
+
+	// Validate the rootPath is consistent with the gopsutil envvars.
+	for _, envVarKey := range []string{
+		"HOST_PROC",
+		"HOST_SYS",
+		"HOST_ETC",
+		"HOST_VAR",
+		"HOST_RUN",
+		"HOST_DEV",
+	} {
+		// If root_path is set, HOST_PROC must also be set.
+		// Other env vars needn't be set, but we check for consistency if they are set.
+		envVarVal := os.Getenv(envVarKey)
+		if envVarVal == "" && envVarKey != "HOST_PROC" {
```

Comment:
Similar to my first comment, it's unclear why we require `HOST_PROC` to be set if rootPath is provided. I believe `rootPath` should always take precedence and set all the gopsutil env vars if they are empty.

Reply:
Fixed it up, as mentioned elsewhere 👍